src/declarations/stencil-public-compiler.ts

```diff
@@ -2192,30 +2192,268 @@ export type OutputTarget =
   | OutputTargetStats
   | OutputTargetDistTypes;
 
+/**
+ * Our custom configuration interface for generated caching Service Workers
+ * using the Workbox library (see https://developer.chrome.com/docs/workbox/).
+ *
+ * Although we are using Workbox we are unfortunately unable to depend on the
+ * published types for the library because they must be compiled using the
+ * `webworker` lib for TypeScript, which cannot be used at the same time as
+ * the `dom` lib. As a workaround the properties are copy-pasted here from
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c7b4dadae5b320ad1311a8f82242b8f2f41b7b8c/types/workbox-build/generate-sw.d.ts#L3
```

Comment:
as I note here we can't at present depend on these types because you need to have the `webworker` lib enabled (and that conflicts with the `dom` lib so both can't be set at the same time 😢).

If we have project references enabled at some time in the future, allowing parts of the codebase to be compiled as separate units, I think we'd be able to carve off the service-worker related code into it's own module and compile it against that lib instead, but for now we'll have to copy-paste this :/